src/estimation/include/iDynTree/Estimation/AttitudeMahonyFilter.h

```diff
@@ -0,0 +1,169 @@
+     * This does not reset the internal state.
+     * @param[in] params object of AttitudeMahonyFilterParameters passed as a const reference
+     * @return true/false if successful/not
+     */
+    bool setParameters(const AttitudeMahonyFilterParameters& params);
+
+    /**
+     * @brief Get filter parameters as a struct.
+     * @param[out] params object of AttitudeMahonyFilterParameters passed as reference
+     */
+    void getParameters(AttitudeMahonyFilterParameters& params);
+
+    bool updateFilterWithMeasurements(const iDynTree::LinearAccelerometerMeasurements& linAccMeas,
+                                      const iDynTree::GyroscopeMeasurements& gyroMeas) override;
+    bool updateFilterWithMeasurements(const iDynTree::LinearAccelerometerMeasurements& linAccMeas,
+                                      const iDynTree::GyroscopeMeasurements& gyroMeas,
+                                      const iDynTree::MagnetometerMeasurements& magMeas) override;
+    bool propagateStates() override;
+    bool getOrientationEstimateAsRotationMatrix(iDynTree::Rotation& rot) override;
+    bool getOrientationEstimateAsQuaternion(iDynTree::Quaternion& q) override;
+    bool getOrientationEstimateAsRPY(iDynTree::RPY& rpy) override;
+    size_t getInternalStateSize() const override;
+    bool getInternalState(iDynTree::Span<double> & stateBuffer) const override;
+    bool getInternalInitialState(iDynTree::Span<double> & stateBuffer) const override;
+    bool setInternalState(iDynTree::Span<double> & stateBuffer) override;
+
+private:
+    AttitudeMahonyFilterParameters m_params;   ///< struct holding the Mahony filter parameters
+
+    /** @struct state internal state of the estimator
+     * @var state::m_orientation
+     * orientation estimate in \f$ \mathbb{R}^4 \f$ quaternion representation
+     * @var state::m_orientation
+     * angular velocity estimate in \f$ \mathbb{R}^3 \f$
+     * @var state::m_orientation
+     * gyroscope bias estimate in \f$ \mathbb{R}^3 \f$
+     */
+    struct {
+        iDynTree::Quaternion m_orientation;
+        iDynTree::Vector3 m_angular_velocity;
+        iDynTree::Vector3 m_gyroscope_bias;
+    } m_state, m_initial_state;
+
+    iDynTree::Rotation m_orientationInSO3; ///< Rotation object to hold orientation estimate as rotation matrix
+    iDynTree::RPY m_orientationInRPY;      ///< RPY object to hold orientation estimate in Euler angles roll-pitch-yaw representation
+
+    iDynTree::Vector3 m_omega_mes; ///< vectorial estimate from accelerometer and magnetometer measurements, \f$ \omega_{mes} \in \mathbb{R}^3 \f$, notation from the paper Non linear complementary filters on the special orthogonal group
+    iDynTree::GyroscopeMeasurements m_Omega_y; ///< gyroscope measurement, \f$ \Omega_{y} \in \mathbb{R}^3 \f$, notation from the paper Non linear complementary filters on the special orthogonal group
+
+    iDynTree::Direction m_gravity_direction; ///< direction of the gravity vector, default set to e3
```

Comment:
What's e3? I think (0, 0, 1.0) would be more readable and consistent with the `iDynTree::Direction` class.

Reply:
Direction in which frame? Inertial?

Reply:
Yes, it's direction in the inertial frame. I will add the details.